src/sentry/web/frontend/newest_performance_issue.py

```diff
@@ -10,19 +10,18 @@
 from .react_page import ReactPageView
 
 
-def get_projects(request, organization):
+def _get_project_ids(request, organization):
     if is_active_superuser(request):
-        return list(
-            Project.objects.filter(status=ProjectStatus.VISIBLE, organization_id=organization.id)
-        )
+        return Project.objects.filter(
+            status=ProjectStatus.VISIBLE, organization_id=organization.id
+        ).values_list("id", flat=True)
     else:
-        return request.access.projects
+        return request.access.visible_project_ids
```

Comment:
I considered whether the changes to the `Access` object may have surpassed this helper function, and that in the `is_active_superuser` case it might already be returning the full set of project IDs.

Sadly, no. The `is_active_superuser` case generally produces an `OrganizationGlobalAccess` object, which (unlike `OrganizationGlobalMembership` for a Sentry app token) goes out of its way *not* to include the full set of projects. It looks as though @scefali wrote this helper function because he wanted the `OrganizationGlobalMembership`-like behavior in this special case. It might be nice to pull this behavior into the `Access` interface somehow, so as not to duplicate the `status=ProjectStatus.VISIBLE` filtering and so on, but I don't know how to do that without making the `Access` interface even more confusing.

It concerns me slightly that there is no test coverage on the `if is_active_superuser(request)`. I might want to pause and add some.

Reply:
Yessir!  This is a good catch, and I recall seeing this similarly.  It was part of the reason I cut back the scope when I realized the special cases and some lack of coverage around superuser.

Let's, not rush this at all.  Let us carefully examine these callsites and consider the correct behavioral refactor.  I'm willing to pair on this more closely.